RabbitMQ.Stream.Client/Message.cs

```diff
@@ -1,29 +1,58 @@
 using System;
 using System.Buffers;
+using System.Collections.Generic;
 using RabbitMQ.Stream.Client.AMQP;
 
 namespace RabbitMQ.Stream.Client
 {
-    public readonly struct Message
+    public struct Hello
+    {
+    }
+
+    public class Message
     {
-        private readonly Properties properties;
-        
         public Message(byte[] data) : this(new Data(new ReadOnlySequence<byte>(data)))
         {
         }
 
-        public Message(Data data, Properties properties = new Properties())
+
+        public Message(Data data)
         {
             this.Data = data;
-            this.properties = properties;
         }
 
+        public Annotations Annotations { get; internal set; }
+
+        public ApplicationProperties ApplicationProperties { get; set; }
+
+        public Properties Properties { get; set; }
         public Data Data { get; }
-        public int Size => Data.Size;
+
+        public int Size => Data.Size +
+                           (Properties?.Size ?? 0) +
+                           (Annotations?.Size ?? 0) +
+                           (ApplicationProperties?.Size ?? 0);
 
         public int Write(Span<byte> span)
         {
-            return Data.Write(span);
+            var offset = 0;
+            if (Properties != null)
+            {
+                offset += Properties.Write(span.Slice(offset));
+            }
+
+            if (ApplicationProperties != null)
+            {
+                offset += ApplicationProperties.Write(span.Slice(offset));
+            }
+
+            if (Annotations != null)
+            {
+                offset += Annotations.Write(span.Slice(offset));
+            }
+
+            offset += Data.Write(span.Slice(offset));
+            return offset;
         }
 
         public ReadOnlySequence<byte> Serialize()
```

Comment:
This method does not seem to be referenced. is it needed?

Reply:
do you mean `Serialize()` ?

Atm is not used yet. At some point we will use it